docs/root/configuration/listeners/udp_filters/dns_filter.rst

```diff
@@ -0,0 +1,61 @@
+.. _config_udp_listener_filters_dns_filter:
+
+DNS Filter
+==========
+
+.. attention::
+
+  DNS Filter is under active development and should be considered alpha and not production ready.
+
+* :ref:`v2 API reference <envoy_api_msg_config.filter.udp.dns_filter.v2alpha.DnsFilterConfig>`
+* This filter should be configured with the name *envoy.filters.udp_listener.dns_filter*
+
+Overview
+--------
+
+The DNS filter allows Envoy to respond to DNS queries as an authoritative server for any configured
+domains. The filter's configuration specifies the names and addresses for which Envoy will answer
+as well as the configuration needed to send queries externally for unknown domains.
+
+The filter supports :ref:`per-filter configuration
+<envoy_api_msg_config.filter.udp.dns_filter.v2alpha.DnsFilterConfig>`.
+An Example configuration follows that illustrates how the filter can be used.
+
+Example Configuration
+---------------------
+
+.. code-block:: yaml
+
+  listener_filters:
+    name: "envoy.filters.udp.dns_filter"
+    typed_config:
+      "@type": "type.googleapis.com/envoy.config.filter.udp.dns_filter.v2alpha.DnsFilterConfig"
+      stat_prefix: "dns_filter_prefix"
+      server_config:
+        control_plane_config:
+          external_retry_count: 3
+          known_domains:
+            - "domain1.com"
+            - "domain2.com"
+            - "domain3.com"
+          virtual_domains:
+            - name: "www.domain1.com"
+              endpoint:
+                addresslist:
```

Comment:
wouldn't this need to be modified to address_list? just curious if the unit test passes in an invalid config

Reply:
also, i didn't find a unit test here. i think that is a requirement for any new additional code since it would not have code coverage otherwise

Reply:
Sure I can update that.

There aren't any tests because the code does nothing other than consume config.  There is nothing that exercises the filter.